File: src/main/java/com/manson/domain/AbstractObject.java
package com.manson.domain;

import com.fasterxml.jackson.annotation.JsonAnyGetter;
import com.fasterxml.jackson.annotation.JsonAnySetter;
import java.util.HashMap;
import java.util.Map;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.ToString;

@Data
@ToString
@EqualsAndHashCode
public abstract class AbstractObject {

    private Map<String, Object> unknownFields = new HashMap<>();

    @JsonAnyGetter
    public final Map<String, Object> getUnknownFieldsMap() {
        return this.unknownFields;
    }

    @JsonAnySetter
    public final void setUnknownField(String key, Object value) {
        this.unknownFields.put(key, value);
    }
}

File: src/main/kotlin/com/manson/AbstractObject.kt
package com.manson

import com.fasterxml.jackson.annotation.JsonAnyGetter
import com.fasterxml.jackson.annotation.JsonAnySetter

open class AbstractObject {
    protected var unknownFields: MutableMap<String, Any> = HashMap()

    @JsonAnyGetter
    fun getUnknownFieldsMap(): MutableMap<String, Any> {
        return unknownFields
    }

    @JsonAnySetter
    fun setUnknownField(key: String, value: Any) {
        unknownFields[key] = value
    }



    override fun toString(): String {
        return "AbstractObject(unknownFields=$unknownFields)"
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is AbstractObject) return false

        if (unknownFields != other.unknownFields) return false

        return true
    }

    override fun hashCode(): Int {
        return unknownFields.hashCode()
    }


}